queue/worker-interaction/index.md

```diff
@@ -0,0 +1,60 @@
+---
+layout:             default
+class:              markdown
+sequence_diagrams:  true
+---
+
+Queue - Worker Interaction (draft)
+==================================
+This document outlines a how queue and worker can interact, and how Azure Queue
+Storage queues participates in this interaction. We refer to TaskCluster Queue
+simply as _queue_, where as Azure Queue Storage queues, will be referred to as
+_Azure queues_.
+
+
+Polling for Tasks
+=================
+The simplest way to poll for tasks is to call `claimWork` on the queue.
+The queue will then poll the underlying Azure queue, and claim a task for you
+if one is available. However, this won't well if a lot of workers are
+constantly polling the queue.
+
+**Proposal,** the queue could track number of `claimWork` requests per 10s, and
+return exponentially large back-off times to workers. Though this is complicated
+to implement.
+
+The efficient way to poll for tasks is to call `pollTaskUrl`. This returns a
```

Comment:
Yeah, it should probably be multiple URLs per workerType... That must be polled in the given order...